app/src/main/java/com/gophillygo/app/activities/EventsMapsActivity.java

```diff
@@ -0,0 +1,96 @@
+package com.gophillygo.app.activities;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.ViewModelProviders;
+import android.content.Intent;
+import android.databinding.DataBindingUtil;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuItem;
+
+import com.google.android.gms.maps.GoogleMap;
+import com.gophillygo.app.R;
+import com.gophillygo.app.data.EventViewModel;
+import com.gophillygo.app.data.models.EventInfo;
+import com.gophillygo.app.data.networkresource.Resource;
+import com.gophillygo.app.data.networkresource.Status;
+import com.gophillygo.app.databinding.FilterButtonBarBinding;
+import com.gophillygo.app.databinding.ActivityEventsMapsBinding;
+
+import java.util.List;
+
+public class EventsMapsActivity extends MapsActivity<EventInfo> {
+
+    private static final String LOG_LABEL = "EventsMapsActivity";
+
+    @SuppressWarnings("WeakerAccess")
+    EventViewModel viewModel;
+
+    public EventsMapsActivity() {
+        super(R.id.events_map, R.id.events_map_toolbar);
+    }
+
+    @Override
+    public void onMapReady(GoogleMap googleMap) {
+        super.onMapReady(googleMap);
+
+        viewModel = ViewModelProviders.of(this, viewModelFactory)
+                .get(EventViewModel.class);
+        LiveData<Resource<List<EventInfo>>> data = viewModel.getEvents();
+        data.observe(this, eventsResource -> {
+            if (eventsResource != null && eventsResource.status.equals(Status.SUCCESS) &&
+                    eventsResource.data != null && !eventsResource.data.isEmpty()) {
+                attractions = eventsResource.data;
+                loadData();
+                // Remove observer after loading full list so updates to the events flags don't
+                // cause unwanted changes to map position
+                data.removeObservers(this);
+            }
+        });
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.events_map_menu, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
```

Comment:
This is largely duplicated between the two maps views, apart from view IDs, and the list view toggle. These should be combined in the base class implementation; we can add an abstract method as one way to handle to the list view toggling.

Also, the TODO for #11 is left in here, but isn't that implemented there now?

Reply:
I considered abstracting this out, but since both the view IDs and the activities they start are different in each activity, I decided against it.